src/Rules/PhpDoc/MethodConditionalReturnTypeRule.php

```diff
@@ -31,7 +31,12 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
-		return $this->helper->check($variants[0]);
+		$variant = $variants[0];
+		$templateTypeMap = $node->getClassReflection()->getTemplateTypeMap()->union(
```

Comment:
`union` isn't the best operation for this. I suggest `array_merge` instead, like here: https://github.com/phpstan/phpstan-src/blob/b250a8bf799b6d22b979260ecfa7539f16449a0f/src/Analyser/NameScope.php#L164-L167